test/perf/script_generation.py

```diff
@@ -68,43 +73,50 @@ def add_dataset_setup(script_lines, perf_test_configuration):
-                            'estimator=estimator, '
-                            'constraints={}())'.format(perf_test_configuration.disparity_metric))
-    elif perf_test_configuration.mitigator == GridSearch.__name__:
-        script_lines.append('mitigator = GridSearch(estimator=estimator, '
-                            'constraints={}())'.format(perf_test_configuration.disparity_metric))
-    else:
-        raise Exception("Unknown mitigation technique.")
-
-    script_lines.append('print("Fitting mitigator")')
-    script_lines.append('mitigator.fit(X_train, y_train, sensitive_features=sensitive_features_train)')
-
-    if perf_test_configuration.mitigator == ThresholdOptimizer.__name__:
-        script_lines.append('mitigator.predict('
-                            'X_test, '
-                            'sensitive_features=sensitive_features_test, '
-                            'random_state=1)')
-    else:
-        script_lines.append('predictions = mitigator.predict(X_test)')
-
-
-def add_evaluation(script_lines):
-    # TODO evaluate accuracy/fairness tradeoff
-
-    script_lines.append('total_time = time() - start_time')
-    script_lines.append('run.log("total_time", total_time)')
-    script_lines.append('print("Total time taken: {}s".format(total_time))')
+    with TimedExecution(_MITIGATION, script_lines):
+        if perf_test_configuration.mitigator == ThresholdOptimizer.__name__:
+            script_lines.append('mitigator = ThresholdOptimizer('
+                                'unconstrained_predictor=unconstrained_predictor, '
+                                'constraints="{}")'.format(perf_test_configuration.disparity_metric))
+        elif perf_test_configuration.mitigator == ExponentiatedGradient.__name__:
+            script_lines.append('mitigator = ExponentiatedGradient('
+                                'estimator=estimator, '
+                                'constraints={}())'.format(perf_test_configuration.disparity_metric))
+        elif perf_test_configuration.mitigator == GridSearch.__name__:
+            script_lines.append('mitigator = GridSearch(estimator=estimator, '
+                                'constraints={}())'.format(perf_test_configuration.disparity_metric))
+        else:
+            raise Exception("Unknown mitigation technique.")
+
+        script_lines.append('mitigator.fit(X_train, y_train, sensitive_features=sensitive_features_train)')
+
+        if perf_test_configuration.mitigator == ThresholdOptimizer.__name__:
+            script_lines.append('mitigator.predict('
+                                'X_test, '
+                                'sensitive_features=sensitive_features_test, '
+                                'random_state=1)')
+        else:
+            script_lines.append('predictions = mitigator.predict(X_test)')
+
+
+def add_additional_metric_calculation(script_lines):
+    # We need to know how much overhead fairlearn adds on top of the basic estimator fit.
```

Comment:
Do we record how many times `estimator.fit()` is called? Because that's part of the measure too.

Reply:
expgrad does that actually! It's called n_oracle_calls and saved in the expgrad result. I can log that! Thanks for the suggestion.

Reply:
…. and record the time for each call (you can't assume that each is the same as the unmitigated estimator)